java/ql/test/experimental/query-tests/security/CWE-600/UncaughtServletException.java

```diff
@@ -0,0 +1,35 @@
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletException;
+
+class UncaughtServletException extends HttpServlet {
```

Comment:
Test cases to add:

* Nested try-blocks
* One top-level method calling another (e.g. implementing `doHead` using `doGet` with post-processing)
* Try-block that rethrows its exception

Some of these might produce false positives / negatives; if so and you think they're too hard to handle, document that fact in the test comment.

Reply:
I've added several more test cases of these three categories. The only one that the current query cannot detect is the test case <code>doOptions</code> with rethrowing unprocessed exceptions containing full stack trace, which is uncommon but is still possible. I've added a note to the test case.